docs/api.yaml

```diff
@@ -0,0 +1,925 @@
+      - $ref: '#/components/parameters/group_id_param'
+    get:
+      description: >-
+        View or Peek message(s) from the DLT associated with this subscription.
+        Consumer pointer is not moved.
+        TODO::
+        - Add pagination semantics to the API.
+      tags: 
+        - Consume
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/recieved_message'
+        default:
+          $ref: '#/components/responses/error_response'
+          
+    put:
+      description: >-
+        Unsideline messages from DLT for further consumption.
+      tags: 
+        - Consume
+      responses:
+        '200':
+          description: OK
+        default:
+          $ref: '#/components/responses/error_response'
+   
+     
+components:
+  schemas: 
+    tenant:
+      type: object
+      properties:
+        name:
+          type: string
+          example: "finance-project"
+        quota:
+          type: object
+          # TODO:: To be enabled later, based on enforcement static/dynamic. 
+          # properties:
+          #   storage:
+          #     type: integer
+          #   throughput:
+          #     type: integer
+        
+    tenants:
+      type: array
+      items:
+        $ref: '#/components/schemas/tenant'
+
+    property_bag:
+      type: object
+      additionalProperties:
+        type: string
+        
+    retention_policy:
+      type: object
+      properties:
+        retention_hours:
+          type: integer
+          example: 24
+          default: 48
+          maximum: 76
+          
+    capacity_policy:
+      type: object
+      properties:
+        max_throughput:
+          type: integer
+          description: "Maximum throughput in Mbps."
+          example: 10
+        max_storage:
+          type: integer
+          description: "Maximum size in MB"
+          example: 1500
+        message_size:
+          type: integer
+          description: "Maximum message size in KBs."
+          example: 10
+          
+    retry_policy:
+      type: object
+      properties:
+        retry_codes:
+          type: array
+          items:
+            type: integer
+          example: [500, 501]
+        kind:
+          type: string
+          example: "exponential"
+          enum: 
+            - "exponential"
+            - "linear"
+        minbackoff:
+          type: integer
+          description: "minimum retry interval in ms."
+          example: 300
+        maxbackoff:
+          type: integer
+          description: "maxmimum retry interval in ms."
+          example: 1500
+        multiplier:
+          type: integer
+          description: "step for iteration"
+          example: 10
+        retry_attempts:
+          type: integer
+          description: "maximum retry attempts allowed."
+          maximum: 3
```

Comment:
so there are "small" / frequent retries that we we can do inline (i.e. during consumption itself), but there are longer retries that we can do using topics.

How do configs map to the above behaviour?

Reply:
As discussed, multiplier is just that, its description may not be be correct. 
So, now the current policy will be backed by topics. 
From implementation point of view, there may be scope of doing some initial retries locally only, before we park such messages in to RQ. do we need to make that explicit as part of policy.